proto/p4/config/p4info.proto

```diff
@@ -55,6 +53,22 @@ message Preamble {
   repeated string annotations = 4;
 }
 
+// used to group all extern instances of the same type in one message
+message Extern {
+  // the extern_type_id is assigned during compilation. It is likely that this
+  // id will in fact come from a P4 annotation to the extern declaration and
+  // that each vendor will receive a prefix to avoid collisions.
```

Comment:
I think we still have the issue of the id being too small for me to be comfortable with the idea. On a related note, do you know if there is a way in P4_16 to declare extern types in a specific namespace? 2 different vendors may wish to both declare a `counter` extern, and even though it is unlikely that they would both be used in the same P4 program, it would be nice to be able to have `vendor1.counter` and `vendor2.counter`.